### PR TITLE
feat: new notification components

### PR DESCRIPTION
**Pull request by matteoscurati**
## **Description**

The notifications team is working to introduce a new type of notifications within the extension. These notifications will be listed in a menu where the user can access basic information. By clicking on an single notification, the user will be able to access detailed data.

This PR introduces the necessary components to display individual items within the list and the various types of details for each notification type.

## **Manual testing steps**

Currently, the individual components can be tested within Storybook. For each created component, there is a corresponding story for testing and demonstration purposes.

### **After**

These are two screenshots taken from the Figma design created by the design team. They represent the notifications list page and the notification detail page.

**Notifications list**
https://www.figma.com/file/c7GgNw2kScGrVyRGAPhwEd/Notifications?type=design&node-id=1030%3A98661&mode=design&t=MHqDKTFq36ifnkqJ-1

**Notification details page**
https://www.figma.com/file/c7GgNw2kScGrVyRGAPhwEd/Notifications?type=design&node-id=1030%3A98983&mode=design&t=MHqDKTFq36ifnkqJ-1

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by matteoscurati** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/components/multichain/notification-detail-nft/index.scss
Comment (darkwing) [inline ui/components/multichain/notification-detail-nft/index.scss:4]: You may be able to use `width` and `height` attributes instead of these ; the height/width are based 4, so 28.

File: ui/components/multichain/notification-detail/notification-detail.tsx
Comment (darkwing) [inline ui/components/multichain/notification-detail/notification-detail.tsx:87]: Should this be `{secondaryTextRight ?? null}`?

File: ui/components/multichain/notification-list-item-text/notification-list-item-text.tsx
Comment (darkwing) [inline ui/components/multichain/notification-list-item-text/notification-list-item-text.tsx:40]: Can this be moved into a helper function?  Seems like a lot to have in a template string.

**Review by darkwing — CHANGES_REQUESTED** [review]
Just a few comments here

File: ui/components/multichain/notification-detail-collection/index.scss
Comment (matteoscurati) [reply ui/components/multichain/notification-detail-collection/index.scss:3]: @darkwing, I might be missing something, but it appears that I cannot use a direct numerical value as a width or height. Instead, I am required to use a value of type BlockSize or BlockSizeArray? Am I right?

File: ui/components/multichain/notification-list-item-text/notification-list-item-text.tsx
Comment (matteoscurati) [reply ui/components/multichain/notification-list-item-text/notification-list-item-text.tsx:40]: You are right about that. Fixed 👍

File: ui/components/multichain/notification-detail-collection/index.scss
Comment (darkwing) [reply ui/components/multichain/notification-detail-collection/index.scss:3]: Oh weird; did this change @georgewrmarshall ?  I thought we could use numerical values.

Comment (georgewrmarshall) [reply ui/components/multichain/notification-detail-collection/index.scss:3]: Apologlies, `width` and `height` props are percentage based. For static pixel values you will have to add the CSS like you have here

Comment (matteoscurati) [reply ui/components/multichain/notification-detail-collection/index.scss:3]: ah ok, thx!

**Comment by matteoscurati** [comment]
@darkwing I had already reviewed your comments and now, due to some conflicts, I have performed a force push.
Do you think it's possible to merge this PR? The components don't affect anything that's in production :)

**Review by darkwing — DISMISSED** [review]
This looks good and I have some non-blocking suggestions:

1.  This top text on the right looks larger than the left, is that wanted?  
SCR-20240227-ljnt

2.  The network badges (on NFTs, for example) display on the bottom right, but all of our UI's badges display on the top right.

**Comment by matteoscurati** [comment]
> This looks good and I have some non-blocking suggestions:
> 
> 1. This top text on the right looks larger than the left, is that wanted?
> 
> SCR-20240227-ljnt
> 2. The network badges (on NFTs, for example) display on the bottom right, but all of our UI's badges display on the top right.

thx @darkwing I'll double check with the design team as soon as possible!

File: ui/components/multichain/notification-list-item-text/notification-list-item-text.tsx
Comment (Prithpal-Sooriya) [inline ui/components/multichain/notification-list-item-text/notification-list-item-text.tsx:37]: Minor note. I don't think we need to "generate" a unique key. Using an index here for the key is fine.

This component won't be re-rendering with its list items in a different order, so should be fine to stick with an index.

This is minor, so you can perform cleanup later.

**Review by Prithpal-Sooriya — DISMISSED** [review]
Notification Dev - Looks good from my perspective.
This PR is completely unconnected & won't effect existing screens/UI.

Treat this Approval as a thumbs up, but wait on review from Extension Team.

**Review by Prithpal-Sooriya — DISMISSED** [review]

**Review by darkwing — APPROVED** [review]

File: ui/components/multichain/notification-list-item/notification-list-item.tsx
Comment (Prithpal-Sooriya) [inline ui/components/multichain/notification-list-item/notification-list-item.tsx:55]: As a cleanup, lets remove all console.logs once we start integrating these components.

Note to others - these components are not connected in any way inside the extension. But will soon be integrated in followup PRs.

**Review by Prithpal-Sooriya — APPROVED** [review]
Nice, lets get this first notifications PR in 🎉

**Pull request merged**
